util/geo.py

```diff
@@ -0,0 +1,160 @@
+import geopy.distance
+import math
+
+from geographiclib.geodesic import Geodesic
+from pyproj import Transformer
+
+MERCATOR_TO_WGS = Transformer.from_crs("EPSG:3857", "EPSG:4326", always_xy=True)
+WGS_TO_MERCATOR = Transformer.from_crs("EPSG:4326", "EPSG:3857", always_xy=True)
+
+def abs_angular_delta(a, b):
+  delta = abs(a - b)
+  return delta if delta <= 180 else 360 - delta
+
+def angle_between_segments(p0, p1, p2):
+  p0lon, p0lat = p0
+  p1lon, p1lat = p1
+  p2lon, p2lat = p2
+
+  azi_a = Geodesic.WGS84.Inverse(p0lat, p0lon, p1lat, p1lon).get('azi2')
+  azi_b = Geodesic.WGS84.Inverse(p1lat, p1lon, p2lat, p2lon).get('azi2')
+  return abs_angular_delta(azi_a, azi_b)
+
+def get_coords(feature):
+  geo = feature.get('geometry', feature)
+  return geo.get('coordinates')
+
+def filter_small_segments(linestring, min_length):
+  coords = get_coords(linestring)
+
+  if len(coords) < 2:
+    return coords
+
+  new_coords = [coords[0]]
+
+  prev = coords[0]
+
+  for i in range(1, len(coords)):
+    cur = coords[i]
+    a = reversed(prev)
+    b = reversed(cur)
+    d = geopy.distance.distance(a, b).meters
```

Comment:
will this function take the reversed a,b iterators or do they need to be converted to lists